secant/Util/ScrollableWhenScaled.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  ScrollableWhenScaled.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 12/27/21.
+//
+
+import SwiftUI
+
+struct ScrollableWhenScaledUpModifier: ViewModifier {
```

Comment:
If you move this into the extension on View below and make it private, that makes the only way it can be used is by accessing the public func you expose.

Reply:
I ended up doing this. Doing what you described above threw numerous errors. I think the result is the same.

````Swift

fileprivate struct ScrollableWhenScaledUpModifier: ViewModifier {
    @ScaledMetric var scale: CGFloat = 1

    func body(content: Content) -> some View {
        if scale > 1 {
            ScrollView {
                content
            }
        } else {
            content
        }
    }
}

extension View {
    func scrollableWhenScaledUp() -> some View {
        self.modifier(ScrollableWhenScaledUpModifier())
    }
}
````

Reply:
```suggestion
import SwiftUI

extension View {
    private struct ScrollableWhenScaledUpModifier: ViewModifier {
        @ScaledMetric var scale: CGFloat = 1

        func body(content: Content) -> some View {
            if scale > 1 {
                return ScrollView {
                    content
                }
            } else {
                return content
            }
        }
    }

    func scrollableWhenScaledUp() -> some View {
        self.modifier(ScrollableWhenScaledUpModifier())
    }
}
```

You just need return statements in an extension

Reply:
this is forbidden by the lint rules

Reply:
I would just go with your fileprivate solution then 👍

Reply:
the fileprivate is forbidden by the linter :(

Reply:
You can comment it out

Reply:
The problem with the suggested change is that the view builders are lost by the nesting or something like that. The compiler is not able to figure out the returning type and infers that they don't match 
![image](https://user-images.githubusercontent.com/484008/149001481-b33a113b-17ba-4efd-98fe-2002980077f4.png)

Reply:
I actually when for the case of declaring fileprivate and adding a swiftlint exception for that next line